### PR TITLE
(CPR-446) Build OpenSSL and CURL standalone

### PR DESCRIPTION
**Pull request by MikaelSmith**
These are useful for AppVeyor CI.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
I believe these still work. They're what I used to produce the packages on https://www.myget.org/F/puppetlabs.

File: configs/components/curl.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+component 'curl' do |pkg, settings, platform|
+  pkg.version '7.46.0'
+  pkg.md5sum '230e682d59bf8ab6eca36da1d39ebd75'
+  pkg.url "http://buildsources.delivery.puppetlabs.net/curl-#{pkg.get_version}.tar.gz"
```
Comment (ehelms) [inline configs/components/curl.rb]: Is there a publicly available place to get this source?

Comment (MikaelSmith) [inline configs/components/curl.rb]: It's a mirror of https://curl.haxx.se/download/curl-7.46.0.tar.gz. We should be using Vanagon's URL rewriting to point to that mirror, not putting it in components. Not sure when I'll get around to fixing that however.

Comment (ehelms) [reply configs/components/curl.rb]: Thanks for the information. I had noticed similarly something like (https://github.com/puppetlabs/pl-build-tools-vanagon/blob/master/configs/components/sysroot.rb#L42) which is not obvious what that tarball even is or where one might could find it outside of the internal Puppetlabs repositories.

Comment (McdonaldSeanp) [reply configs/components/curl.rb]: @ehelms We generate that sysroot ourselves I believe.

/cc @ScottGarman

Comment (ScottGarman) [reply configs/components/curl.rb]: @ehelms when it comes to sysroot generation, you should be able to find scripts in https://github.com/puppetlabs/pl-build-tools-vanagon/tree/master/scripts that can be used to try to re-create those tarballs (generally from a minimal install of the OS). For platforms such as RHEL and SLES I think there may be licensing issues to making those public. If you wanted access to the Debian/Ubuntu ones, let me know and I can share them with you if wanted.

File: configs/components/openssl.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+component "openssl" do |pkg, settings, platform|
+  pkg.version "1.0.2d"
```
Comment (shrug) [inline configs/components/openssl.rb]: This is an old openssl version, current is 1.0.2l

**Comment by MikaelSmith** [comment]
Updated to include mirrors and use the latest version.

**Comment by shrug** [comment]
Apparently I really really approve

**Pull request merged**